File: Desafio1/desafio1.py
def adicionar_aluno(nome, lista_alunos):
    """Adiciona um nome à lista de alunos, caso ainda não esteja nela.
    str, list -> None"""
    
    nome_formatado = " ".join(
        palavra.capitalize() if i == 0 or len(palavra) > 3 else palavra
        for i, palavra in enumerate(nome.lower().split())
    )
    
    if nome_formatado in lista_alunos:
        print("Aluno já está na lista.")
    else:
        lista_alunos.append(nome_formatado)


def exibir_lista(lista):
    """Exibe todos os alunos cadastrados.
    list -> None"""
    print("\nAlunos cadastrados:")
    for i, aluno in enumerate(lista, 1):
        print(f"{i}. {aluno}")

def main():
    """Gerencia a interação do usuário para cadastrar alunos.
    None -> None"""
    lista_alunos = []

    while True:
        try:
            nome = input("Digite o nome do aluno: ")
            if nome.lower() == "sair":
                break
            adicionar_aluno(nome, lista_alunos)
        except Exception as e:
            print(f"Erro: {e}. Tente novamente.")
    
    exibir_lista(lista_alunos)

if __name__ == "__main__":
    main()

File: Desafio1/desafio1.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class desafio1 {

    public static void adicionarAluno(String nome, List<String> listaAlunos) {
        String nomeFormatado = formatarNome(nome);
        if (listaAlunos.contains(nomeFormatado)) {
            System.out.println("Aluno já está na lista.");
        } else {
            listaAlunos.add(nomeFormatado);
        }
    }

    public static String formatarNome(String nome) {
        String[] palavras = nome.toLowerCase().split(" ");
        StringBuilder nomeFormatado = new StringBuilder();

        for (int i = 0; i < palavras.length; i++) {
            if (i == 0 || palavras[i].length() > 3) {
                nomeFormatado.append(Character.toUpperCase(palavras[i].charAt(0)))
                             .append(palavras[i].substring(1));
            } else {
                nomeFormatado.append(palavras[i]);
            }
            if (i < palavras.length - 1) {
                nomeFormatado.append(" ");
            }
        }

        return nomeFormatado.toString();
    }

    public static void exibirLista(List<String> listaAlunos) {
        System.out.println("\nAlunos cadastrados:");
        for (int i = 0; i < listaAlunos.size(); i++) {
            System.out.println((i + 1) + ". " + listaAlunos.get(i));
        }
    }

    public static void main(String[] args) {
        List<String> listaAlunos = new ArrayList<>();
        Scanner scanner = new Scanner(System.in);

        while (true) {
            try {
                System.out.print("Digite o nome do aluno: ");
                String nome = scanner.nextLine();
                if (nome.equalsIgnoreCase("sair")) {
                    break;
                }
                adicionarAluno(nome, listaAlunos);
            } catch (Exception e) {
                System.out.println("Erro: " + e.getMessage() + ". Tente novamente.");
            }
        }

        exibirLista(listaAlunos);
        scanner.close();
    }
}